.github/workflows/dev_carbon-aware-api.yml

```diff
@@ -1,7 +1,7 @@
 # Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
 # More GitHub Actions for Azure: https://github.com/Azure/actions
 
-name: Build and deploy ASP.Net Core app to Azure Web App - carbon-aware-api
+name: 2.a-Build and deploy ASP.Net Core app to Azure Web App - carbon-aware-api

```

Comment:
Should we deploy the app from the container built by 2-pre-release.yaml ? You mentioned about this before:
https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/347#issuecomment-1538378214
> can we update it to deploy from the newly deployed package (if triggered in DEV, using latest "pre" tagged image, and if from MAIN, using "latest" tagged image) please?

We can work for this in another issue of course.

Reply:
Yes we should but I was trying to at least get this through. I can see that even like that I will need to improve further (some tasks repeat but trying to keep the changes low risk so it can be reviewed more easily), but I now have dependencies so if a workflow that describes a foundation steps for others fail, it can just stop there instead of running all workflows even when the solution does not compile...

Can you help review so we can get that through and merge with your changes to get them through too please? @YaSuenag 

(so deploy will be a different PR, but as you can see I already had a sample that I commented out in the end)

cc @Willmish @vaughanknight

Reply:
My main problem was DCO - had some conflicts on my local but will go back to that. In the meantime, please start reviewing